util/image_viewer/spotfinder_frame.py

```diff
@@ -823,12 +824,24 @@ def sum_images(self):
             ).index
             imageset = self.image_chooser.GetClientData(i_frame).image_set
 
-            for i in range(1, self.params.sum_images):
+            for i in range(1, self.params.stack_images):
                 if (i_frame + i) >= len(imageset):
                     break
                 image_data_i = imageset[i_frame + i]
                 for j, rd in enumerate(image_data):
-                    rd += image_data_i[j]
+                    data = image_data_i[j]
+                    if mode == "max":
+                        if type(rd) != type(data):
```

Comment:
I think you can call `.as_double()` even on an array that is already `flex.double()`, so you could avoid having to explicitly check the type and always both sides to double.

`a = flex.random_double(10).as_double()`